t/ui/10-tests_overview.t

```diff
@@ -607,6 +607,19 @@ subtest "job dependencies displayed on 'Test result overview' page" => sub {
       'parent job was highlighted correctly';
 };
 
+subtest 'result filter does not affect scheduled and running jobs' => sub {
```

Comment:
I didn't look close enough - I think this test belongs in `t/ui/01-list.t`.
In `t/ui/10-tests_overview.t` we are testing `/tests/overview`, and your added test is the only one testing the `/tests` "all tests" page.